Decred Wallet/Features/Send/SendViewController.swift

```diff
@@ -503,42 +503,47 @@ extension SendViewController {
     }
     
     func checkAddressFromQrCode(textScannedFromQRCode: String?) {
-        guard var capturedText = textScannedFromQRCode else {
+        
+        guard let capturedText = textScannedFromQRCode else {
             self.destinationAddressTextField.text = ""
             return
         }
         
-        if capturedText.starts(with: "decred:") {
-            capturedText = capturedText.replacingOccurrences(of: "decred:", with: "")
-        }
+        let URI: DecredAddressURI = DecredAddressURI(uriString: capturedText)
         
-        if capturedText.count < 25 {
-            self.invalidAddressFromQrCode(errorMessage: LocalizedStrings.walletAddressShort)
-            return
-        }
-        if capturedText.count > 36 {
-            self.invalidAddressFromQrCode(errorMessage: LocalizedStrings.walletAddressLong)
+        let address = URI.address
+        
+        if address.count < 25 || address.count > 36 {
+            self.invalidAddressFromQrCode(errorMessage: String(format: LocalizedStrings.addressFromQr, address), title: LocalizedStrings.invalidAddr)
             return
         }
         
         if BuildConfig.IsTestNet {
-            if capturedText.starts(with: "T") {
-                self.destinationAddressTextField.text = capturedText
+            if address.starts(with: "T") {
+                self.destinationAddressTextField.text = address
             } else {
                 self.invalidAddressFromQrCode(errorMessage: LocalizedStrings.invalidTesnetAddress)
             }
         } else {
-            if capturedText.starts(with: "D") {
-                self.destinationAddressTextField.text = capturedText
+            if address.starts(with: "D") {
+                self.destinationAddressTextField.text = address
             } else {
                 self.invalidAddressFromQrCode(errorMessage: LocalizedStrings.invalidMainnetAddress)
             }
         }
+        // Only fill amount if address is valid and QR code contains an amount.
+        let addressValid = self.sendFromWallet.isAddressValid(self.destinationAddressTextField.text)
+        if addressValid && URI.amount != nil {
```

Comment:
We don't check if the address is valid before displaying it in the address text field, why are we checking it's validity before displaying the amount.

Also, it is possible that a valid address exists in `self.destinationAddressTextField.text` but that is not the address from the qr code.

Reply:
refactored to only check for amount returned from QR code in `b2c1b1f`